kyaml/setters2/set.go

```diff
@@ -226,6 +226,19 @@ func (s *Set) set(field *yaml.RNode, ext *CliExtension, sch *spec.Schema) (bool,
 		return true, nil
 	}
 
+	// If we don't know the desired type of the value we are setting,
+	// verify that the value is valid for the tag of the node.
+	if len(sch.Type) == 0 {
+		isValid, err := yaml.ValidValueForTag(ext.Setter.Value, field.YNode().Tag)
```

Comment:
Not sure if this is going to work -- some nodes can be IntOrString

Reply:
But IntOrString is not a valid yaml type right? So even if the schema type for a field is IntOrString, the actual type of the yaml node will be either `!!int` or `!!string`. If we have additional information about the type, either from the openapi schema or from the user (through the `--type` parameter in create-setter), I think we can do better. This PR just tries to address the situation where we have no information about the type of the field.

Reply:
I could be wrong on this, but I believe it is possible to have a field which may be *either* an int *or* a string -- and both which specifies the field as an int andyaml which specifies the field a string are both valid.

Would you confirm this is the case?

You can test by setting maxUnavailable: https://github.com/kubernetes/api/blob/63b8484e576699c2f714dd6319c827a568a95090/apps/v1/types.go#L358

https://godoc.org/k8s.io/apimachinery/pkg/util/intstr#Parse

Reply:
So the schema allows the field to have either string or int value. But at any time, the actual value will be one of the two options, and the type of the node when we work with it in kyaml will be either `!!int` or `!!string`. We just have to make sure that when we change the value of the node, we either make sure that the new value is still valid for the type of the node, or we have to change the tag of the node to reflect the new type.

This PR is meant to only cover the situation where we don't have type information for the field, so no openapi schema and no type provided by the user. We might be able to do better with more type information, but with union types (like IntOrString) it might still be difficult to make good decisions on which type we should choose. 

I have verified the use case where the schema for a field has the type IntOrString. If we create a setter for the field maxUnavailable with an initial value of 2, and then try to update the value to 30%, the result will be `!!int 30%`. Similar, if the initial value of the field is 30% and we use a setter to update the value to 4, the result ends up as `"4"`.

Reply:
I have updated this PR with a new approach that will address this. So it will be possible to create a setter for `maxUnavailable` and set it to either an integer or a string.